pkg/server/server_test.go

```diff
@@ -1383,3 +1383,33 @@ func TestDefaultMaxConcurrentReadSettings(t *testing.T) {
 	require.EqualValues(t, math.MaxUint32, s.maxConcurrentReadsForCheck)
 	require.EqualValues(t, math.MaxUint32, s.maxConcurrentReadsForListObjects)
 }
+
+func TestWriteAuthorizationModelWithExperimentalEnableModularModels(t *testing.T) {
+	ctx := context.Background()
+	storeID := ulid.Make().String()
+
+	mockController := gomock.NewController(t)
+	defer mockController.Finish()
+
+	mockDatastore := mockstorage.NewMockOpenFGADatastore(mockController)
+
+	s := MustNewServerWithOpts(
+		WithDatastore(mockDatastore),
+	)
+
+	t.Run("rejects_request_with_schema_version_1.2", func(t *testing.T) {
+		mockDatastore.EXPECT().MaxTypesPerAuthorizationModel().Return(100)
+
+		_, err := s.WriteAuthorizationModel(ctx, &openfgav1.WriteAuthorizationModelRequest{
+			StoreId:       storeID,
+			SchemaVersion: typesystem.SchemaVersion1_2,
+			TypeDefinitions: []*openfgav1.TypeDefinition{
+				{
+					Type: "user",
+				},
+			},
+		})
+
+		require.ErrorIs(t, err, status.Error(codes.InvalidArgument, "modular models (schema version 1.2) is not supported"))
+	})
+}
```

Comment:
can you also make this a unit test in the `commands` pkg?

```go
func TestWriteAuthorizationModelWithExperimentalEnableModularModels(t *testing.T) {
	ctx := context.Background()
	storeID := ulid.Make().String()

	mockController := gomock.NewController(t)
	defer mockController.Finish()
	mockDatastore := mockstorage.NewMockOpenFGADatastore(mockController)

	testCases := map[string]struct {
		enableModularModules bool
		inputSchema          string
		expectAllowed        bool
	}{
		`allow`: {
			enableModularModules: true,
			inputSchema:          typesystem.SchemaVersion1_2,
			expectAllowed:        true,
		},
		`forbid`: {
			enableModularModules: false,
			inputSchema:          typesystem.SchemaVersion1_2,
			expectAllowed:        false,
		},
	}

	for testName, test := range testCases {
		t.Run(testName, func(t *testing.T) {
			cmd := commands.NewWriteAuthorizationModelCommand(mockDatastore).
				WithWriteAuthModelEnableModularModels(test.enableModularModules)
			_, err := cmd.Execute(ctx, &openfgav1.WriteAuthorizationModelRequest{
				StoreId: storeID,
				TypeDefinitions: []*openfgav1.TypeDefinition{
					{
						Type: "user",
					},
				},
				SchemaVersion: test.inputSchema,
			})
			if test.expectAllowed {
				require.NoError(t, err)
			} else {
				require.ErrorIs(t, err, status.Error(codes.InvalidArgument, "modular models (schema version 1.2) is not supported"))
			}
		})
	}
}
```

the test you wrote we will keep it to test that the actual server flag is passed downstream

Reply:
Added into 4245759b5bd252234c557bfed49c7ca6ff5f4321